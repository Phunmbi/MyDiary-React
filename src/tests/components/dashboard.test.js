import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import DashboardComponent, { Dashboard } from '../../components/dashboard';
import Root from '../../root';

let wrapper;
let wrapped;

const props = {
  getAllEntries: jest.fn(),
  entries: {
    entries: {
      entries: {
        allEntries: {
          status: 200,
          data: {
            length: 0,
            message: 'new',
          },
        },
      },
    },
  },
};

document.getElementById = jest.fn((selector) => {
  console.log('selector => ', selector);

  return {
    style: {
      background: 'background',
    },
    innerHTML: 'inner',
    disabled: true,
  };
});

beforeEach(() => {
  jest.useFakeTimers();
  wrapper = mount(
    <Root>
      <MemoryRouter initialEntries={[{ key: 'testkey' }]}>
        <DashboardComponent
          getAllEntries={props.getAllEntries}
          response={props.entries}
        />
      </MemoryRouter>
    </Root>,
  );

  wrapped = shallow(
    <Dashboard
      getAllEntries={props.getAllEntries}
      response={props.entries}
    />,
  );
});

afterEach(() => wrapper.unmount());

describe('Dashboard UI', () => {
  describe('render features', () => {
    test('component should render as expected', () => {
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
    });
  });
});

describe('Dashboard Functionality', () => {
  it('should click simulate submit', () => {
    const inst = wrapped.instance();

    inst.props = {
      addEntry: jest.fn(),
      match: {
        params: {
          id: 1,
        },
      },
      response: {
        entries: {
          entries: {
            allEntries: {
              status: 200,
              data: {
                length: jest.fn(),
                message: 'new',
              },
            },
          },
        },
      },
    };

    inst.forceUpdate();
  });
});
