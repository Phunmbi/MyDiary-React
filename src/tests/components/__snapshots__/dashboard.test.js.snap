// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard UI render features component should render as expected 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <MemoryRouter
      initialEntries={
        Array [
          Object {
            "key": "testkey",
          },
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "key": "testkey",
                "pathname": "/",
                "search": "",
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "testkey",
              "pathname": "/",
              "search": "",
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(Dashboard)
          getAllEntries={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          response={
            Object {
              "entries": Object {
                "entries": Object {
                  "allEntries": Object {
                    "data": Object {
                      "length": 0,
                      "message": "new",
                    },
                    "status": 200,
                  },
                },
              },
            }
          }
        >
          <Dashboard
            getAllEntries={[Function]}
            response={
              Object {
                "auth": Object {
                  "authenticated": null,
                  "firstName": null,
                  "lastName": null,
                },
                "entries": Object {
                  "authenticated": "",
                  "deletedEntry": Array [],
                  "deletedEntryError": Array [],
                  "deletedReminder": Array [],
                  "deletedReminderError": Array [],
                  "editedEntry": Array [],
                  "editedEntryError": Array [],
                  "entries": Array [],
                  "entriesError": Array [],
                  "error": Array [],
                  "existingReminder": Array [],
                  "existingReminderError": Array [],
                  "firstName": "",
                  "lastName": "",
                  "newEntry": Array [],
                  "newEntryError": Array [],
                  "newReminder": Array [],
                  "newReminderError": Array [],
                  "singleEntry": Array [],
                  "singleEntryError": Array [],
                  "status": "",
                },
                "profile": Object {
                  "authenticated": "",
                  "deletedEntry": Array [],
                  "deletedEntryError": Array [],
                  "deletedReminder": Array [],
                  "deletedReminderError": Array [],
                  "editedEntry": Array [],
                  "editedEntryError": Array [],
                  "entries": Array [],
                  "entriesError": Array [],
                  "error": Array [],
                  "existingReminder": Array [],
                  "existingReminderError": Array [],
                  "firstName": "",
                  "lastName": "",
                  "newEntry": Array [],
                  "newEntryError": Array [],
                  "newReminder": Array [],
                  "newReminderError": Array [],
                  "singleEntry": Array [],
                  "singleEntryError": Array [],
                  "status": "",
                },
              }
            }
          >
            <div
              className="dashboard"
            >
              <Header>
                <header>
                  <Link
                    replace={false}
                    to="/dashboard"
                  >
                    <a
                      href="/dashboard"
                      onClick={[Function]}
                    >
                      <h1>
                        MyDiary
                      </h1>
                    </a>
                  </Link>
                  <Link
                    replace={false}
                    to="/profile"
                  >
                    <a
                      href="/profile"
                      onClick={[Function]}
                    >
                      <img
                        alt="Profile"
                        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+P+/HgAFhAJ/wlseKgAAAABJRU5ErkJggg=="
                      />
                    </a>
                  </Link>
                </header>
              </Header>
              <section>
                <h3>
                  Dashboard
                </h3>
              </section>
              <div
                id="loadingModal"
              >
                <div
                  className="loadingModal-main"
                >
                  <img
                    alt="Loading"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAEBAQAAACH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  />
                </div>
              </div>
              <div
                className="add"
              >
                <Link
                  replace={false}
                  to="/entries/add"
                >
                  <a
                    href="/entries/add"
                    onClick={[Function]}
                  >
                    <span>
                      +
                    </span>
                  </a>
                </Link>
              </div>
            </div>
          </Dashboard>
        </Connect(Dashboard)>
      </Router>
    </MemoryRouter>
  </Provider>
</_default>
`;
